tests/operator/operator.go

```diff
@@ -3035,7 +3036,36 @@ spec:
 
 				By("Checking launcher seccomp policy")
 				vmi := libvmi.NewCirros()
-				vmi = tests.RunVMIAndExpectScheduling(vmi, 60)
+				vmi = tests.RunVMI(vmi, 60)
+				fetchVMI := matcher.ThisVMI(vmi)
+				psa := false
+				err := wait.PollImmediate(time.Second, 30*time.Second, func() (done bool, err error) {
+					vmi, err := fetchVMI()
+					if err != nil {
+						return done, err
+					}
+
+					if vmi.Status.Phase != v1.Pending {
+						return true, nil
+					}
+
+					for _, condition := range vmi.Status.Conditions {
+						if condition.Type == v1.VirtualMachineInstanceSynchronized {
+							if condition.Status == k8sv1.ConditionFalse && strings.Contains(condition.Reason, "needs a privileged namespace") {
+								psa = true
+								return true, nil
+							}
+						}
+					}
+					return
+				})
+				Expect(err).NotTo(HaveOccurred())
+				// It is expected to see error on clusters with PSA enabled. The Pod below can't be executed.
```

Comment:
I don't completely understand the comment here. If an error is expected on PSA enabled machines, will the test then not directly fail? And should the `if` statement not get moved before the `Expect` above?

Reply:
In the case of PSA the above poll will confirm that we see the expected error in the condition. The error returned from the poll can be only a network issue or VMI disappearing.